### PR TITLE
feature: added the ngx.ssl.session module

### PR DESCRIPTION
**Pull request by agentzh**
This Lua module provides Lua APIs for the contexts ssl_session_fetch_by_lua* and ssl_session_store_by_lua*.

See https://github.com/openresty/lua-nginx-module/pull/822 for more details.

### CODE REVIEW

**Comment by agentzh** [comment]
@ghedo @lziest @doujiang24 @yangshuxin Please review this PR. Thanks!

File: lib/ngx/ssl/session.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+            -- cache miss...just return
+            return
+        end
+
+        local ok, err = ssl_sess.set_serialized_session(sess)
+        if not ok then
+            ngx.log(ngx.ERR, "failed to set SSL session for ID ", sess_id,
+                    ": ", err)
+            -- considier it as a cache miss...
```
Comment (lziest) [inline lib/ngx/ssl/session.md]: typo "consider"

File: lib/ngx/ssl/session.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+                    "current connection: ", err)
+            -- just give up
+            return
+        end
+
+        -- for the best performance, we should avoid creating a closure
+        -- dynamically here on the hot code path. Instead, we should
+        -- put this function in one of our own Lua module files. this
+        -- example is just for demonstrative purposes...
+        local function save_it(premature, sess_id, sess)
```
Comment (yangshuxin) [inline lib/ngx/ssl/session.md]: > for demonstrative purposes:

for *demonstration" purpose

File: lib/ngx/ssl/session.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+        if not sess_id then
+            ngx.log(ngx.ERR, "failed to get session ID: ", err)
+            -- considered a cache miss, and just return...
+            return
+        end
+
+        -- the user is supposed to implement the my_lookup_ssl_session_by_id
+        -- Lua function used below. She can look up an external memcached
+        -- or redis cluster, for example. And she can also introduce a local
+        -- cache layer at the same time...
```
Comment (yangshuxin) [inline lib/ngx/ssl/session.md]: IMHO, the above comment is not very straightforward, I would say "It is up to the user to implement xyz() function, which is to retrieve the session corresponding to the given id".

**Comment by agentzh** [comment]
This PR has just been merged. New issues found should be reported as separate GitHub issues. Thanks!

**Pull request merged**